resolvers.py

```diff
@@ -0,0 +1,1979 @@
+        # 2. start date, end date
+        # 3. type of data (Available, Nominated, Allocation Pending, Total bench)
+        # 4. band name (Optional)updated_end_date__range=(start_date, end_date),
+
+        capability_name = kwargs.get("capability_name")
+        band_name = kwargs.get("band_name")
+        start_date = kwargs.get("start_date")
+        end_date = kwargs.get("end_date")
+        bench_type = kwargs.get(
+            "bench_type"
+        )  # Allocation Pending, Nominated, Available, Total Bench
+
+        qs_filters = Q(nomination_status__in=bench_type.value)
+
+        if None in bench_type.value:
+            qs_filters = qs_filters | Q(nomination_status__isnull=True)
+
+        qs = models.BaseBenchDataSource.objects.filter(qs_filters)
+        qs = qs.filter(updated_end_date__range=(start_date, end_date))
+        qs = qs.filter(band__in=BENCH_REPORTS_BAND,
+                       on_bench__in=ON_BENCH, allocation_end_dates__isnull=True)
+        qs = qs.filter(department='Delivery')
+
+        # validate capability_name
+        if str(capability_name).lower() == "total":
+            return qs
+
+        if str(capability_name).lower() == "other":
+            # handle case like "Java,QA"
+            qs = qs.filter(capabilities=None)
+        else:
+            qs = qs.filter(capabilities=capability_name)
+
+        if band_name:
+            qs = qs.filter(band=band_name)
+
+        return qs
+
+
+class TotalBaseDataSourceResolverByCapability:
+    @custom_resolver_check_perm
+    def __call__(self, source, info, **kwargs):
+        # params required
+        # 1. capability name
+        # 2. type of data (Available, Nominated, Allocation Pending, Total bench)
+
+        capability_name = kwargs.get("capability_name")
+        bench_type = kwargs.get("bench_type")
+
+        qs_filters = Q(nomination_status__in=bench_type.value)
+
+        if None in bench_type.value:
+            qs_filters = qs_filters | Q(nomination_status__isnull=True)
+
+        qs = models.BaseBenchDataSource.objects.filter(qs_filters)
+        qs = qs.filter(updated_end_date__isnull=False)
+        qs = qs.filter(band__in=BENCH_REPORTS_BAND,
+                       on_bench__in=ON_BENCH, allocation_end_dates__isnull=True)
+        qs = qs.filter(department='Delivery')
+
+        if str(capability_name).lower() == "other":
+            # handle case like "Java,QA"
+            return qs.filter(capabilities=None)
+        elif str(capability_name).lower() == "total":
+            return qs
+
+        return qs.filter(capabilities=capability_name)
+
+
+class BaseDataSourceResolverByBand:
+    @custom_resolver_check_perm
+    def __call__(self, source, info, **kwargs):
+        # params required
+        # 1. band name
+        # 2. start date, end date
+        # 3. type of data (Available, Nominated, Allocation Pending, Total bench)
+
+        band_name = kwargs.get("band_name")
+        start_date = kwargs.get("start_date")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

You are using the `get` method without a default argument to return the value of a key in a dictionary. We recommended that you use a default argument so that if the value for your key is not found, a default value is returned. If a default value is not provided and the key is not found, then `None` is returned.

[Learn more](https://docs.python.org/3/library/stdtypes.html#dict.get)

Similar issue at line number 369.